*** /home/ftp/looking-glass/mrlg-5.1.0/archive-contents/index.cgi	2005-01-18 08:37:17.000000000 -0500
--- index.cgi	2005-01-21 22:10:45.000000000 -0500
*************** no warnings "once";
*** 8,10 ****
  
! $::Version='5.1.0 (IPv6+SSH)';
  
--- 8,10 ----
  
! $::Version='5.1.1 (IPv6+SSH)';
  
*************** $::Version='5.1.0 (IPv6+SSH)';
*** 24,25 ****
--- 24,42 ----
  ##
+ ## This file is part of the Multi-Router Looking Glass (MRLG).
+ ##
+ ## Multi-Router Looking Glass (MRLG) is free software; you can redistribute
+ ## it and/or modify it under the terms of the GNU General Public License as
+ ## published by the Free Software Foundation; either version 2, or (at 
+ ## your option) any later version.
+ ##
+ ## Multi-Router Looking Glass (MRLG) is distributed in the hope that it will
+ ## be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+ ## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General 
+ ## Public License for more details.
+ ##
+ ## You should have received a copy of the GNU General Public License
+ ## along with Multi-Router Looking Glass (MRLG); see the file LICENSE.  If not,
+ ## write to the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ ## Boston, MA 02111-1307, USA.
+ ##
  ## Please leave copyright notices in tact.  If for whatever reason, you 
